docs/src/lib/metadata.md

```diff
@@ -149,24 +150,25 @@ data frames.
     Such changes might be made based on users' feedback about what metadata
     propagation rules are most convenient in practice.
 
-The general design rules for propagation of table and column level metadata
-are as follows:
-* if some object (data frame or column) is mutated in-place its metadata
-  is not changed.
-* preferably (this is not guaranteed in all cases as sometimes it might be hard
-  to detect that this condition is met) if some object is moved as a whole
-  (either by coping or by aliasing) then its metadata is propagated.
-* in all other cases if a new object is allocated from an existing object and
-  this transformation is not a copy or aliasing then metadata is dropped.
+The general design rules for propagation of table metadata is as follows:
+* for all operations that take a single data frame like object
+  and return a data frame like object table level metadata is preserved;
+* for all operations that take more than one data frame like object
+  and return a data frame like object (e.g. `hcat` or joins) table level metadata
+  is preserved from all tables with the exception that if two or more tables
+  define the same metadata then it is silently dropped.
```

Comment:
Maybe this rule is not conservative enough? For example if one dataset has `"source" => "Labour Force Surveys"` and the other one has no source information, saying that the result of a join comes from the LFS would be wrong or exaggerated. Same for observation year or license. We could preserve only metadata which is identical in all inputs.

Reply:
> We could preserve only metadata which is identical in all inputs.

OK - we can start being more conservative and loosen it later.

Reply:
I have thought more about it. I would keep what I propose (drop metadata only if it is inconsistent across tables).
The point is that if we followed your suggestion then we would need to drop metadata on any change of data (e.g. adding a column or changing values in the column by `setindex!` - and we would go back to my original "super conservative" rules).

I agree that the current rule can lead to exaggerated metadata, but the other alternative is "no mutation" rule I proposed earlier, which was overly conservative - right?

Reply:
I'd say the situation of combining multiple data frames can be considered as different from mutating a single data frame (which can be thought of as changing only the contents of the table). The possibility of conflicts doesn't even exist in the latter case.

BTW, I realize that we cannot really drop metadata on any mutation, as one can mutate vectors directly without DataFrames functions being involved.

Reply:
> I realize that we cannot really drop metadata on any mutation

Yes, but this is user's choice to do this.

> I'd say the situation of combining multiple data frames can be considered as different

OK - I will then update the rule that only metadata that is the same for all tables is kept.